amethyst_network/src/metrics/mod.rs

```diff
@@ -0,0 +1,59 @@
+//! Module for managing network metrics.
+
+use serde::Serialize;
+use std::time::{Duration, Instant};
+mod console_metrics;
+mod csv_metrics;
+mod influxdb_metrics;
+mod network_metrics;
+
+pub use self::console_metrics::ConsoleMetrics;
+pub use self::csv_metrics::CsvMetrics;
+pub use self::influxdb_metrics::InfluxDBMetrics;
+pub use self::network_metrics::NetworkMetrics;
+
+/// Metrics observer which could be implemented to observe metrics changes.
+/// This trait could be implemented for handling metrics a certain way.
+pub trait MetricObserver<T: Sync + Send> {
+    /// This method will be called to when a change in some metric has occurred.
+    fn update(&mut self, metrics_object: &T);
```

Comment:
This could take a `delta` of the elapsed time since the last update?

Reply:
Interesting I'll take a look